docs/query_proxy.md

```diff
@@ -0,0 +1,205 @@
+
+#### Creating New API Keys
+
+Each user must have an API key. These keys only have meaning to the proxy server. They are not passed to the guardians.
+The proxy requires that a key be present in each query request, and that the specified key exists in the permissions file.
+Beyond that, the API keys have no special meaning. They can be generated using a site like [this](https://www.uuidgenerator.net/version4).
+
+#### Updating the Permissions File
+
+The proxy server monitors the permissions file for changes. Whenever a change is detected, it reads the file, validates it, and if
+it passes validation, switches to the new version. Care should be taken when editing the file while the proxy server is running, because
+as soon as you save the file, the changes will be picked up (whether they are logically complete or not).
+
+## Telemetry
+
+The proxy server provides two types of telemetry data, logs and metrics.
+
+### Logging
+
+The proxy server uses the same logging mechanism as the guardian. It will write to a local file, but can also be configured to
+publish logs to Grafana using the Loki protocol. If you will be running your proxy server in mainnet, you should contact foundation
+personnel about getting a Grafana ID to be used for logging and use it to set the `--telemetryLokiURL` command line argument.
+
+If you set the log level to `info`, the proxy server logs information on all incoming requests and output bound responses. This can
+be helpful for determining when requests reach quorum, but may be too chatty as the level of queries traffic grows. If that is the
+case, you can set the log level to `warn`.
+
+### Metrics
+
+The proxy server uses Prometheus to track various activity and can publish them to Grafana. If you will be running your proxy server in mainnet,
+you should contact foundation personnel about getting a Grafana ID and use it to set the `--promRemoteURL` command line argument.
+
+For the set of available metrics, see [here](../node/cmd/ccq/metrics.go).
+
+## Troubleshooting
+
+### P2P Health
+
+If you think you are having trouble with your access to the P2P network, you can add `--monitorPeers` to the command line arguments,
+which will cause the proxy server to periodically check its connectivity to the P2P bootstrap peers, and attempt to reconnect if necessary.
+
+### Invalid Requests
+
+If the proxy server determines that a request is invalid, it does the following:
+
+- Logs an error message using the user name (not the API Key).
+- Pegs the appropriate Prometheus metric.
+- Sends a failure response to the user.
+
+Note that if the proxy server thinks a request is valid, but the guardians do not, the guardians silently drop the request, so it will look
+like a timeout. This is to avoid a denial of service attack on the guardians. This can happen if the proxy server is not properly provisioned
```

Comment:
```suggestion
like a timeout. This is to avoid a denial of service attack on the guardians. This can happen if the proxy server is not properly permissioned
```